src/sql/workbench/services/connection/browser/media/connectionDialog.css

```diff
@@ -111,6 +111,25 @@
 	font-weight: 600;
 }
 
+.custom-connection-options .connection-label .dialog-label-container{
+	background-image: url('info_notification.svg');
+	background-size: 9px;
+	background-repeat: no-repeat;
+	background-position: right;
+	padding-right: 18px;
+	float: left;
+}
+
+.vs-dark .custom-connection-options .connection-label .dialog-label-container,
+.hc-black .custom-connection-options .connection-label .dialog-label-container {
```

Comment:
Instead of making a duplicate icon it would be better to move the modal notification icon to src\sql\media\icons and add it to common-icons.css

(you can still keep connection dialog specific properties here as needed since common-icons should just be the background-image value)